### ISSUE TITLE
Thunderbird segfaults on NixOS

### ISSUE THREAD

**Issue by neilmayhew** [NixOS/nixpkgs#65563]
**Describe the bug**

Starting `thunderbird` (from `thunderbird` or `thunderbird-bin`) segfaults.

**To Reproduce**

From a terminal, or the GUI, run `thunderbird`, eg:

```shellsession
$ thunderbird -ProfileManager -safe-mode
Segmentation fault
```

**Expected behavior**

Doesn't crash.

**Screenshots**

Not applicable.

**Additional context**

I'm running on NixOS from the `unstable-small` channel. I've created a new profile and I'm running TB in safe mode. The segfault doesn't happen until after the first account is created.

Using `gdb` I can see that it's deeply nested inside `libxul` when the crash happens.

This started happening consistently after I did a channel update and `nixos-rebuild`, so I used `git bisect` to find the commit(s) that caused the problem. If I revert these two commits, the problem goes away:

* `90b1197301e` nixos/xdg: add portal option
* `00eef848b29` nixos/doc: fix build

(The second commit is needed only to fix a build failure in the documentation that was introduced by the first commit.)

I have to log out or reboot for the change to take effect when using `switch` with `nixos-rebuild`.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.134, NixOS, 19.09.git.72e547e (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre186865.6016ec50c3f"`
 - channels(user): `"nixpkgs-19.09pre186574.88d9f776091"`
 - nixpkgs: `~/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by alapshin** [comment]
I had similar problem recently and looks like it is caused by changes introduced in
https://github.com/NixOS/nixpkgs/commit/d734750608841852b121446bb7791cc4e224c257#diff-857277ef9cb10083de0170c682ae6fe7

Now `xdg.portal.enable` is enabled automatically if there is `services.xserver.enable` in system config, `xdg.portal.enable = true`  in turn sets environment variable `GTK_USE_PORTAL=1`.

And this environment variable causes thunderbird to crash on start:
https://bugs.launchpad.net/ubuntu/+source/thunderbird/+bug/1813471

According to comments in https://bugs.launchpad.net/ubuntu/+source/thunderbird/+bug/1813471/comments/4 this bug is fixed in Thunberbird 68 but it is still in beta at the moment.

As a temporary workaround one could disable xdg-desktop-portal completely using
`xdg.portal.enable = false` in config. Or if this functionality needed elsewhere then one could create custom script that clears this variable before launching thunderbird.

**Comment by neilmayhew** [comment]
Setting `xdg.portal.enable = true` fixes it. Thanks so much for this!

Do you think we should leave this issue open until 68 arrives, so others can see the workaround?

**Comment by alapshin** [comment]
> Do you think we should leave this issue open until 68 arrives, so others can see the workaround?

Yes, I think it would be better to leave it open for now.

**Comment by JosephLucas** [comment]
In config.nix, setting `xdg.portal.enable = false;`, logging out, then logging in solved the issue for me.
You can check the value of the environment variable with `echo $GTK_USE_PORTAL`.

**Comment by gloaming** [comment]
I don't see why we should wait for 68 to release - the current version is a nightly ("daily"), so a beta should be _more_ stable if anything.

**Comment by alapshin** [comment]
> I don't see why we should wait for 68 to release - the current version is a nightly ("daily"), so a beta should be more stable if anything.

I think `Daily` naming suffix is result of disabled official branding in  Thunderbird builds: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/mailreaders/thunderbird/default.nix#L18 
Real version is still ESR 60.8.0 as far as I can tell:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/mailreaders/thunderbird/default.nix#L27

I also tried to build current beta version of Thunberbird 690b1, but
1. There are some changes and new dependencies in upstream build which will require backport of build logic from Firefox
2. Some existing dependencies (`NSS` or `rust-cbinding` for example) require versions that are not yet packaged for NixOS. Maybe this is only for Thunderbird 69 and 68 uses older versions, didn't have a chance to check it yet.

**Comment by gloaming** [comment]
No, that can't be right. This is what it tells me every time I open it:

![thunderbird-daily](https://user-images.githubusercontent.com/10156748/62228420-88f94200-b3b5-11e9-96e4-bccea7bb677a.png)

It's also clearly branded "**Thunderbird** Daily" everywhere. Probably this version simply doesn't support debranding, because being "Daily" already overrides the branding.

**Comment by gloaming** [comment]
@alapshin Thanks for looking into it. If we can't yet use 68, then we should issue a hotfix via a wrapper ASAP. This should also have a backport IMO.

**Comment by sternenseemann** [comment]
Note that the workaround does not work for Plasma and GNOME, since `xdg.portal.enable` is set to `true` there and can't be overridden. I also tried setting `GTK_USE_PORTAL=0` before running thunderbird, which unfortunately does not work.
